frontend/pkg/frontend/cluster.go

```diff
@@ -244,11 +242,24 @@ func (f *Frontend) createHCPCluster(writer http.ResponseWriter, request *http.Re
 		TrackedResource: arm.NewTrackedResource(resourceID),
 	}
 	newExternalCluster.Normalize(newInternalCluster)
+
+	// set fields that were not included during the conversion, because the user does not provide them or because the
+	// data is determined live on read.
+	newInternalCluster.SystemData = systemData
+	// Clear the user-assigned identities map since that is reconstructed from Cluster Service data.
+	// TODO we'd like to have the instance complete when we go to validate it.  Right now validation fails if we clear this.
+	// TODO we probably update validation to require this field is cleared.
+	//newInternalCluster.Identity.UserAssignedIdentities = nil
```

Comment:
I imagine this will get cleaner once the RP itself is fetching client and principal IDs for user-assigned identities instead of depending on Cluster Service to do it.  That might be a good addition to the initial set of tasks for the new backend.